### PR TITLE
Add directionAnchor prop to allow right-aligned Components

### PR DESCRIPTION
**Pull request by travisbloom**
DateRangePicker and SingleDatePicker have a new prop, directionAnchor
that will align the components to the left/right side of the page.
This prop has two new constants, DIRECTION_ANCHOR_LEFT (default)
and DIRECTION_ANCHOR_RIGHT.

solves issue https://github.com/airbnb/react-dates/issues/71

### CODE REVIEW

File: css/DateRangePicker.scss [hunk 1]
```diff
@@ -61,13 +68,19 @@
...
+.DateRangePicker__picker--direction-right.DateRangePicker__picker--start::before {
+  right: 239px;
+}
 
-.DateRangePicker__picker--end::before {
+.DateRangePicker__picker--direction-left.DateRangePicker__picker--end::before {
   left: 162px;
 }
+.DateRangePicker__picker--direction-right.DateRangePicker__picker--end::before {
+  right: 101px;
```
Comment (travisbloom) [inline css/DateRangePicker.scss]: unsure if there's a measurement strategy that should be used to calculate these offsets, just tried to eye them. Let me know if there's a calculation you were relying on (and the we can add a `$sass-variable`)

**Comment by travisbloom** [comment]
@majapw fixes added

**Comment by Olliepop** [comment]
What's necessary to see this merged? This would be huge for us. Do you need any help?

**Comment by majapw** [comment]
Looking into it today! I think it's basically ready. :)

**Comment by travisbloom** [comment]
@majapw just fixed a rebase issue with my other PR, let me know if there are any concerns :)

**Comment by majapw** [comment]
I don't really know why the approval check wasn't passing. :/

**Comment by Olliepop** [comment]
Thanks guys, this is awesome! Is this worthy of a version bump?

**Comment by majapw** [comment]
Just released v3.3.0 which should include this change! :)

**Pull request merged**